app/models/article.rb

```diff
@@ -351,6 +351,22 @@ def update_score
                    spaminess_rating: BlackBox.calculate_spaminess(self))
   end
 
+  def liquid_tags_used(section = nil)
+    content =
+      case section
+      when :body
+        body_markdown
+      when :comments
+        comments_blob
+      else
+        "#{body_markdown}#{comments_blob}"
+      end
+
+    MarkdownParser.new(content).tags_used
+  rescue StandardError
+    []
+  end
```

Comment:
This is a refactor. For the context of the liquid tag, we don't want to check the comments of an article, only the body of the article itself.